R/forms.R

```diff
@@ -369,19 +405,28 @@ relocateForm <- function(formId, newDatabaseId) {
 #' schema
 #' @param databaseId the id of the database to which the form should belong.
 #' @param label the label of the new form
-#' @param folderId the id of the folder where the form should reside; defaults to the database id. This argument only has an effect if upload is TRUE.
+#' @param folderId Deprecated; use parentId. Optional id of a folder where the form should reside. This argument only has an effect if upload is TRUE.
 #' @param keyColumns a character vector of the column names of the form fields that should be form keys
 #' @param requiredColumns a character vector of the column names of the form fields that should be required
 #' @param logicalAsSingleSelect by default TRUE and converts logical columns in the data frame to a single select form field; if FALSE then it will convert TRUE to 1 and FALSE to 0
 #' @param logicalText the single select replacement values for c(TRUE, FALSE); default is c("True","False")
 #' @param codes a character vector of field codes that must have the same length as the number of columns
+#' @param parentId The id of the database or folder to which this should be added. Defaults to the database. This argument only has an effect if upload is TRUE.
+#' @param parentFormId The parent form id when creating a sub-form.
 #' @param upload immediately upload the new form
+#' @param parentIdColumn Indicates the sub-form data column that references the parent form to be ignored in the schema creation
+#'
 #' @export
-createFormSchemaFromData <- function(x, databaseId, label, folderId = databaseId, keyColumns = character(), requiredColumns = keyColumns, logicalAsSingleSelect = TRUE, logicalText = c("True","False"), codes = rep(NA_character_, ncol(x)), upload = FALSE) {
+createFormSchemaFromData <- function(x, databaseId, label, folderId, keyColumns = character(), requiredColumns = keyColumns, logicalAsSingleSelect = TRUE, logicalText = c("True","False"), codes = rep(NA_character_, ncol(x)), upload = FALSE, parentId = folderId, parentFormId = NULL, parentIdColumn = NULL) {
```

Comment:
rename `parentIdColumn` to `parentRecordIdColumn`

Reply:
I noted an inconsistency. In importRecords we use parentIdColumn but I think we should rename this to be parentRecordIdColumn to match and to be consistent also with submitPending() and addRecord(). Since it would be a breaking change, I'll just deprecate the old parameter.